arguments/SubgroupOps.argdown

```diff
@@ -6,30 +6,48 @@ Expose a subset of subgroup operations in WGSL shaders:
-  + Metal has [threadExecutionWidth](https://developer.apple.com/documentation/metal/mtlcomputepipelinestate/1414911-threadexecutionwidth) to query the size.
-  - D3D12 doesn't have anything to query the exact size.
-  - <Useless>: it's not actionable in Vulkan and D3D12, unlike Metal where we can control the workgroup size from the API.
-
-[Quad Ops]: We should expose the quad operations in fragment shaders.
-  - new Adreno and PowerVR hardware doesn't support this
-//TODO: fill this out more
-
-[Non-uniform]: We should support the non-uniform subgroup model.
-  + <Everywhere>: Vulkan, D3D, and Metal use a non-uniform model.
-  - <Ambiguous Divergence>: when invocations in a subgroup are executing "together", how long is that guaranteed?
-  - <Ambiguous Reconvergence>: once invocations diverge, what are the guarantees about where you reconverge?
-    + Vulkan has weak guarantees, D3D12 and Metal don't have anything.
-    - we can just say invocations never reconverge, which matches AMD ISA and CUDA models.
-  - <Ambiguous Forward Progress>: how blocks affect progress of other blocks, and invocations within blocks affect progress on other invocations?
-    + D3D, Metal, and Vulkan are silent on both of these.
-  - <Ambiguous Helpers>: do helper invocations participate in subgroup operations?
+[Motivation]: This PR works towards issue #667,
+              subgroup operations are common
+              across all three APIs.
+  + <Performance>: Proven contribution to
+                   general purpose algorithms.
+  + <Safe>: There is a safe subset of
+            subgroup operations.
+
+
+[Extension]: Subgroup operations should be
+             exposed as an extension.
+  + <Target>: Subgroup operations are not available
+              on all WebGPU target hardware.
+
+[Device Only]: Host statistics for subgroup
+               operations are not common
+               across all three APIs.
+
+[Compute Only]: Subgroup operations should only
+                work in compute kernels.
+  + <Hardware>: Restricting to compute only increases
+                market (e.g., Adreno).
+  + <Definition>: Operations are better defined
+                  for compute. And this makes helper
+                  invocations irrelevant.
+
+[No Shuffle Operations]: Shuffle operations should be excluded.
+  + <DirectX>: DirectX doesn't have shuffle operations.
+  + <Support>: Not having shuffle operations
+               increases support (e.g., ARM).
+
+[No Quad Operations]: Quad operations should be their
+                      own extension.
+  + <Market>: If quad operations are made into their
+              own extension, both its potential market
+              becomes larger and subgroup operations'
+              market grows.
+
+[No Explicit Operations]: Indexed or masked operations
```

Comment:
here, again, you are using the statement description for the argument. Let's not do that:) The description should be saying "We shouldn't expose explicit broadcast operations" or something like that